src/event/quic/ngx_event_quic_protection.c

```diff
@@ -29,10 +32,10 @@ static uint64_t ngx_quic_parse_pn(u_char **pos, ngx_int_t len, u_char *mask,
     uint64_t *largest_pn);
 
 static ngx_int_t ngx_quic_crypto_open(ngx_quic_secret_t *s, ngx_str_t *out,
-    u_char *nonce, ngx_str_t *in, ngx_str_t *ad, ngx_log_t *log);
+    const u_char *nonce, ngx_str_t *in, ngx_str_t *ad, ngx_log_t *log);
```

Comment:
The const patch does feel incomplete For example, `static ngx_quic_md_t  key` could also be const, but it's not. I feel like the complete patch could be much bigger than that. However, since it only constifies `nonce`, it's ok to commit.

Reply:
Unlike `nonce`, constifying `ngx_quic_md_t` or `ngx_quic_md_t::data` looks less easy & useful, because it is not directly passed to the library and breaks in various places.

OTOH, making `key` just static in ngx_quic_create_retry_packet() is good enough, because this eliminates few load/store instructions by moving it into the data segment.

In principle, we could leave `nonce` just static as well; making it also `const` moves from the data segment to the read-only data segment. I made it `const` because this can be done at no cost.